ui/src/components/schemas/SchemaDetails.tsx

```diff
@@ -185,7 +186,10 @@ export function SchemaDetails() {
                   <Button
                     icon={<CreditCardIcon />}
                     onClick={() => {
-                      navigate(generatePath(ROUTES.issueCredential.path, { schemaID }));
+                      navigate({
+                        pathname: generatePath(ROUTES.issueCredential.path),
+                        search: schemaID ? `?${SCHEMA_SEARCH_PARAM}=${schemaID}` : "",
```

Comment:
This should be

`search: schemaID ? new URLSearchParams(SCHEMA_SEARCH_PARAM, schemaID) : undefined`. (The `?` is never needed even when setting a literal)

If you search for "search: `" you will find other verbatim instances that should be changed to this.

Reply:
That doesn't work as you said. I'm just using `new URLSearchParams` when I want to persist other values, but that method need as well the `set` and after to`toString` like this example
```
const search = new URLSearchParams(searchParams);

      search.set(SCHEMA_SEARCH_PARAM, schema.id);

      navigate(
        {
          pathname: generatePath(ROUTES.issueCredential.path),
          search: search.toString(),
        },
        { replace: true }
      );
```

Reply:
Oops, you are right, I was conflating the signature of the instantiation with the `.set` method.

NB we still don't need the "?".

Reply:
We still need it because the `schemaID` is not checked in the render. Typescript doesn't know that if we have `schema` is because we have `schemaID`

Reply:
Also NB that this still holds:

> `SchemaDetails` should return an `ErrorResult` if `schemaID` is undefined. Then we don't need these subsequent checks.

Reply:
It's not possible to load that page without `schemaID`

Reply:
Sorry to be late to the party but I prefer Adri's original approach here. As Adri mentioned, this page won't be shown if `schemaID` is not defined, so adding this JSX feels misleading:

```typescript
if (!schemaID) {
  return <ErrorResult error="No schema ID provided." />;
}
```

This will never render, but now we have an "impossible error msg" that we need to maintain and what is worst, that induces developers to think this is a legit scenario, while it's not.

I prefer something closer to the original implementation, which is, condition the navigation to the existence of the value:

```typescript
onClick={() => {
  if (schemaID) {
    navigate({
      pathname: generatePath(ROUTES.issueCredential.path),
      search: `${SCHEMA_SEARCH_PARAM}=${schemaID}`,
    });
  }
}}
```

Reply:
I don't understand how this is better. In both cases, the absence of the`schemaID` is practically impossible, being a consequence of the return type of `useParams`. In the return DOM approach, if somehow this were to happen, at least we provide a human-friendly failsafe rather than corrupted data and non-functional interactions.

Is there a way to convince the compiler that could these avoid manual safeguards altogether?

Reply:
> if somehow this were to happen, at least we provide a human-friendly failsafe rather than corrupted data and non-functional interactions

You have a point there. Let's leave it like this.